File: languages/java/src/main/java/com/bloock/sdk/entity/Keys.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.RecordOuterClass.GenerateKeysResponse;

public class Keys extends KeyPair {
  Keys(String publicKey, String privateKey) {
    super(publicKey, privateKey);
  }

  public static Keys fromProto(GenerateKeysResponse keys) {
    return new Keys(keys.getPublicKey(), keys.getPrivateKey());
  }
}

File: languages/js/src/entity/keys.ts
import * as keyProto from "../bridge/proto/keys";

export class KeyPair {
  publicKey: string;
  privateKey: string;

  constructor(publicKey: string, privateKey: string) {
    this.publicKey = publicKey;
    this.privateKey = privateKey;
  }
}

export class EcdsaKeyPair extends KeyPair {
  static fromProto(k: keyProto.GenerateLocalKeyResponse): KeyPair {
    return new KeyPair(k.localKey!.key, k.localKey!.privateKey!);
  }
}

export class RsaKeyPair extends KeyPair {
  static fromProto(k: keyProto.GenerateLocalKeyResponse): KeyPair {
    return new KeyPair(k.localKey!.key, k.localKey!.privateKey!);
  }
}

export class EciesKeyPair extends KeyPair {
  static fromProto(k: keyProto.GenerateLocalKeyResponse): KeyPair {
    return new KeyPair(k.localKey!.key, k.localKey!.privateKey!);
  }
}
